src/GitHub.Api/Primitives/TheVersion.cs

```diff
@@ -47,6 +47,8 @@ private void Initialize(string version)
             if (initialized)
                 return;
 
+            Guard.ArgumentNotNullOrWhiteSpace(version, nameof(version));
+
```

Comment:
Throwing here isn't useful, deserialization might lead to a null string and there's no way of telling without catching exceptions. Just return if it's null or empty.